### PR TITLE
Pagination component (used in the datatable)

### PR DESCRIPTION
**Pull request by zagovorichev**
https://oat-sa.atlassian.net/browse/TAO-3260

### CODE REVIEW

**Comment by bugalot** [comment]
Hi @krampstudio , Hi @zagovorichev ,

I had the opportunity to functionally test this PR in the context of an MP ART PR that relies on this one (https://github.com/oat-sa/extension-tao-mpart/pull/338). The component behaved very well and could not find any defect.

File: views/js/test/ui/pagination/test.js [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016  (original work) Open Assessment Technologies SA;
+ *
+ * @author Alexander Zagovorichev <zagovorichev@1pt.com>
+ */
+
+define(['jquery', 'lodash', 'ui/pagination', 'jqueryui'], function ($, _, paginationComponent) {
```
Comment (krampstudio) [inline views/js/test/ui/pagination/test.js]: I guess the dependency to `jqueryui` is wrong

File: views/js/ui/pagination/providers/pages.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    'lodash',
+    'i18n',
+    'tpl!ui/pagination/providers/tpl/pages'
+], function ($, _, __, tpl) {
+    'use strict';
+
+    var $paginationTpl;
+
+    var generatePage = function generatePage(page) {
+        return $('<li class="page"><span>' + page + '</span></li>');
```
Comment (krampstudio) [inline views/js/ui/pagination/providers/pages.js]: Html code (everything bigger than a tag should be in a template)

File: views/scss/inc/_pagination.scss [hunk 3]
```diff
@@ -0,0 +1,109 @@
+//
+// Pagination (multiple pages)
+// --------------------------------------------------
+.pagination {
+  .pages {
+    display: flex;
```
Comment (krampstudio) [inline views/scss/inc/_pagination.scss]: Please use the mixin from the `function.scss` to generate the version with vendor prefixes.

File: views/scss/inc/_pagination.scss [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+// Large
+.pagination-lg {
+  .pages {
+    > li {
+      display: inline; // Remove list-style and block-level defaults
+      > a,
+      > span {
+        padding: 10px 25px;
+        font-size: 20px;
```
Comment (krampstudio) [inline views/scss/inc/_pagination.scss]: Please use the mixin `@include font-size` instead

File: views/scss/inc/_pagination.scss [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+// Small
+.pagination-sm {
+  .pages {
+    > li {
+      display: inline; // Remove list-style and block-level defaults
+      > a,
+      > span {
+        padding: 5px 18px;
+        font-size: 12px;
```
Comment (krampstudio) [inline views/scss/inc/_pagination.scss]: Please use the mixin `@include font-size` instead

**Comment by krampstudio** [comment]
Unit tests runs successfully and there's no regression on other tables.

**Comment by zagovorichev** [comment]
@krampstudio Update

**Comment by krampstudio** [comment]
Unit tests runs successfully and there's no regression on other tables.

**Pull request merged**